### ISSUE TITLE
'pangyp' is not recognized as an internal or external command,

### ISSUE THREAD

**Issue by adriatic** [zewa666/aurelia-node#1]
Just discovered this repository and since I have some plans related to aurelia-node-mongo, wanted to see what is this about :-). My context is Windows 8.1 and the command `npm install` resulted with the error stated in the title of this issue report. Here is the log from the start to this error message at line 31; (installation continued seemingly unperturbed)

```
λ npm install
npm WARN package.json aurelia-node@0.1.0 No repository field.
npm WARN deprecated exec@0.2.0: use child_process.execFile instead
npm WARN engine karma@0.12.31: wanted: {"node":"~0.8 || ~0.10"} (current: {"node":"0.12.0","npm":"2.5.1"})
npm WARN deprecated 6to5-core@2.13.7: 6to5 is now 'babel': https://babeljs.io/blog/2015/02/15/not-born-to-die
npm WARN deprecated 6to5@2.13.7: 6to5 is now 'babel': https://babeljs.io/blog/2015/02/15/not-born-to-die
npm WARN optional dep failed, continuing fsevents@0.3.5
\


> ws@0.5.0 install C:\work\Learning\aurelia\aurelia-node\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)


C:\work\Learning\aurelia\aurelia-node\node_modules\browser-sync\node_modules\socket.io\node_modules\engine.io\node_modules\ws {git}
{lamb} node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
/
> ws@0.4.31 install C:\work\Learning\aurelia\aurelia-node\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)


C:\work\Learning\aurelia\aurelia-node\node_modules\browser-sync\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws {git}
{lamb} node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
-


> kexec@1.1.0 install C:\work\Learning\aurelia\aurelia-node\node_modules\karma-6to5-preprocessor\node_modules\6to5\node_modules\kexec
> pangyp configure build

'pangyp' is not recognized as an internal or external command,
operable program or batch file.
npm WARN optional dep failed, continuing kexec@1.1.0
npm WARN optional dep failed, continuing fsevents@0.3.5
/


> ws@0.4.32 install C:\work\Learning\aurelia\aurelia-node\node_modules\karma\node_modules\socket.io\node_modules\socket.io-client\node_modules\ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)


C:\work\Learning\aurelia\aurelia-node\node_modules\karma\node_modules\socket.io\node_modules\socket.io-client\node_modules\ws {git}
```

**Comment by zewa666** [comment]
I've honestly never heard about that error. But as far as I understood the example still continues to install and works? NPM likes to through out a lot of errors during install, depending on the console you're invoking it from. Which one was it for you? Next time please try out the Git Bash and see whether you get the same. From my experience that one seems to be much more battle proof in combination with NPM

**Comment by adriatic** [comment]
I am using Cmder (http://bliker.github.io/cmder/) which is vastly superior to any "shells" on Windows. I will also try this same project creation with Git Bash, as I have the suspicion that the problem is in the area of "unix on windows" emulation (Cygwin). 
**RANT**
This points out to my biggest fear related to using the MEAN stack on Windows. Regardless of the obvious fact that this is orders of magnitude better environment than any others, if there are problems with it on Windows, it should not be perceived as "Windows is too feeble" but rather than the npm system was "ported with hate". Cygwin should not be a part of such port - npm should work natively if MEAN developers care about the fact that windows is by far the most used OS **by end-users**. This whole concept of software development is not about developers - it is about people who will benefit from that process by using it.
**END RANT**

**Comment by adriatic** [comment]
It is likely that my RANT does not apply in this case: after reading the https://github.com/TooTallNate/node-gyp description I realized that my version of Python (3.4.2) is not supported by node-gyp, so this may also affect the pangyp.
Please do let me know what version of Python do I need to have on my machine in order to successfully install Aurelia-node - or should I rather wait for a more stable project?

**Comment by zewa666** [comment]
Well as said that's deep in the NPM process chain, so honestly I don't see that as part of this application. Fixing it would certainly be helpful, but I had similar warnings and never experienced any issues so far.
Whether this project is stable or not ... I dunno you decide, it's about giving an idea and helping you with the first steps. I'm sure there is a lot that could be done better, but on the other hand even worse :)

This is OSS so feel free to take whatever you need or discard it at all

**Comment by adriatic** [comment]
My motivation is different from what you may perceived: I am deeply committed to Aurelia (for many reasons), so as a solid perfectionist, I cannot be at peace, seeing how my build process signals a failure in a mid-point and then happily use the result of the build. 
For example: I uninstalled Python 3.x and replaced it with Python 2.x only to see that Aurelia-node `npm install` **fails completely**. So, it seems that just that `pangyp` (which is deemed as a temporary replacement for node-gyp (why??)) requires Python 2.x while the rest of npm suite of tools expects the 3.x version.
I will revert back to python 3.x and the whole purpose of my writing about this issue was to give you the warning how there is something fishy. Feel free to deal with this as you see fit :-) (Ja se ne zelim petljati u Vas posao, jer sam samo posmatrac u ovom slucaju)

**Comment by zewa666** [comment]
I fully get that, the problem is though that I think this is not the place to fix it. Like you mentioned the root problem seems to be with how they ported over NPM to Windows. I guess filling that stuff in their repos might lead to better results. Nevertheless I appreciate your hint about it and I'm sure whoever else has this issue at least now knows he is not alone :)

(Nema frke slobodno recite sta god mislite. To je pojam od OSS i tako i treba da bude.)

**Comment by adriatic** [comment]
Two important points to conclude this discussion - both for the benefit of the whole Aurelia team:
1. Aurelia success has to be unconditional, meaning that it cannot depend on the stability of the foundation that Aurelia team does not control. By virtue of using npm (which was a decision made by Aurelia team) all of you need to ensure to never have to use the explanation "Sorry, this is not our problem, it is node-gyp that inserted a bug in our run-time".
2. More importantly, the fact that OSS is being evolved without pay should never be used in the context "take it or leave it" - the manufacturer has the same responsibility to the user as in any other producer - consumer relationship. 

For the sake of our own relationship, Vildan, please make sure to interpret my text as I wrote it -- to help with my nearly 40 years of working as a software developer or software development process manager.

**Comment by zewa666** [comment]
Thanks for the hints. Also as a closing remark, this is not an official Aurelia Repo, just my private testing and sharing. Nevertheless definitely a virtue we all should strive for :)

**Comment by adriatic** [comment]
:-)

It would be great if someone with sufficient knowledge of npm states the python requirements for "Aurelia development process"

**Comment by adriatic** [comment]
Since the ‘exec’ module seems to be deprecated I wonder whether you would consider updating your “build” to use “child_process.execFile” so I can try whether this works for me.

**Comment by zewa666** [comment]
Hey please try creating a folder public in the root of your project and run the terminal as admin. Lets see if that helps first.

**Comment by adriatic** [comment]
Once I made the `public` folder as you suggested, the `node bin/install' command resulted with:

```
C:\Work\Learning\aurelia\aurelia-node>node bin/install
https://nodeload.github.com/aurelia/skeleton-navigation/zip/master
exec: use child_process.execFile instead
undefined
```

and the newly created folder is populated:

![public folder created](https://cloud.githubusercontent.com/assets/2712405/6582970/47af79aa-c737-11e4-9fab-f3d7b17cd381.png)

Further, running `gulp watch` in public folder results with:

![running gulp](https://cloud.githubusercontent.com/assets/2712405/6583155/a33aa1e0-c738-11e4-9233-dfbb48ede5b8.png)

and

![browser exec](https://cloud.githubusercontent.com/assets/2712405/6583181/cca09ab2-c738-11e4-9f84-7520dea67858.png)

Note sure what should be the app's URL - so this might all be good. 
**Note that I did this test on my Windows 2012 Server - not the Windows 8.1 workstation which still resists to cooperate with me**

**Comment by zewa666** [comment]
Cool, youd have to add /app to the path to go to the APP. So it does work. Nevertheless ill take a look how to work around that warning.

**Comment by adriatic** [comment]
There are still some "gotchas" as I do not know all requirements for Aurelia development. For example running the `http://localhost:7000/app` resulted with the complaint that `config.js` is missing. So, by guessing I issued `jspm install system -g`  which resulted with

```
    Configuration file ..\config.js doesn't exist, create it? [yes]:
         Creating registry cache...
    err  Repo jspm:system not found!
    warn Installation changes not saved.
```

As this is not a good opportunity to bug you with teaching me how to setup the environment - and mode importantly because **we eliminated the doubts on "Python interference"** we can really close this issue. I will come back to ask you for the right setup once I really learn all I need to know how to use "express" :-)
